### ISSUE TITLE
SecRuleUpdateTargetById does not match regular expressions in 3.0

### ISSUE THREAD

**Issue by anush-cr** [SpiderLabs/ModSecurity#1872]
I have not had success with SecRuleUpdateTargetById when using regular expressions. For example, these didn't seem to work (the rules still trigger):

```
SecRuleUpdateTargetById 921151 "!REQUEST_COOKIES:/mixpanel$/"

SecRuleUpdateTargetById 921151 "!REQUEST_COOKIES:/^_hp2_/"
```
I'm using modsecurity along with nginx.

**Comment by victorhora** [comment]
This demands further investigation. Might be related with #1590 and in turn https://github.com/SpiderLabs/ModSecurity-nginx/issues/74

**Comment by j0k2r** [comment]
Hi,

Same problem here with the version 3.0.3.

The rule is ignored when i set the full request cookie name:
```
SecRuleUpdateTargetById 942100 "!REQUEST_COOKIES:PrestaShop-aa8515103a21db7d23b568838ffd034e"
```

But it still triggered when using the regex:

```
SecRuleUpdateTargetById 942100 "!REQUEST_COOKIES:/^PrestaShop-.*/"
```

**Comment by zimmerle** [comment]
Hi @j0k2r and @anush-cr,

Thank you for the report. This issue is currently under investigation. 

I am also looking at the behavior of SecRuleUpdateTargetByTag which does not seems to be working as expected either.

**Comment by zimmerle** [comment]
Thank you for the report the issue is closed as of: 1b8d69da02419664af719f628bf7d0a6cb13daaf

**Comment by kanitan** [comment]
I have the same problem when using SecRuleUpdateTargetById.It seems fixed in v3.0.4 with the commit: https://github.com/SpiderLabs/ModSecurity/commit/1b1fdc055b8071ad3b24573abfe9b96e546c7abf

But actually it still doesn't works.Here is my config (libmodsecurity v3.4.0):
```
SecRuleUpdateTargetById 942100 !REQUEST_COOKIES:/^_id/
```
Send a request with ``Cookie: _idbnaa=16778240#217623#kNDxUN8UsVvg--gXz8eKfQ`` and it was blocked by ruleId 942100.

I confirmed that string match like ``SecRuleUpdateTargetById 942100 !REQUEST_COOKIES:_idbnaa`` is still avalible.
